docs/pages/build/advanced-credentials-configuration.md

```diff
@@ -0,0 +1,262 @@
+
+### Local Mode
+
+You can configure EAS Builds so that `expo eas:build` will be reading the credentials only from `credentials.json`. Just set `"credentialsSource": "local"` in one of your [build profiles](../eas-json/) in `eas.json`.
+
+Example:
+
+```json
+{
+  "builds": {
+    "android": {
+      "release": {
+        "workflow": "generic",
+        "credentialsSource": "local"
+      }
+    }
+  }
+}
+```
+
+This can be particularly useful when running the build on a CI. If the `credentialsSource` is set to `local` but the `credentials.json` file doesn't exist Expo CLI will throw an error.
+
+### Remote Mode
+
+Alternatively, you can choose to always use the remote credentials, even if the `credentials.json` file exists in the project root. To do so, set `"credentialsSource": "remote"` in the build profile.
+
+Example:
+
+```json
+{
+  "builds": {
+    "android": {
+      "release": {
+        "workflow": "generic",
+        "credentialsSource": "local"
+      }
+    }
+  }
+}
+```
+
+## Running Builds on a CI
+
+The concept of the `credentials.json` file facilitates the process of building your React Native project on a CI. The example below shows one of the many ways of automating the release process.
+
+Before you start working on setting up your CI job, make sure you have these two files:
+
+- `credentials.json` with paths and secrets for your credentials, something like this:
+
+  ```json
+  {
+    "android": {
+      "keystore": {
+        "keystorePath": "android/keystores/release.keystore",
+        "keystorePassword": "paofohlooZ9e",
+        "keyAlias": "keyalias",
+        "keyPassword": "aew1Geuthoev"
+      }
+    },
+    "ios": {
+      "provisioningProfilePath": "ios/certs/profile.mobileprovision",
+      "distributionCertificate": {
+        "path": "ios/certs/dist-cert.p12",
+        "password": "iex3shi9Lohl"
+      }
+    }
+  }
+  ```
+
+- `eas.json` ([learn more](../eas-json/)) with build profiles which enforce using the local `credentials.json` file:
+
+  ```json
+  {
+    "builds": {
+      "android": {
+        "release": {
+          "workflow": "generic",
+          "credentialsSource": "local"
+        }
+      },
+      "ios": {
+        "release": {
+          "workflow": "generic",
+          "credentialsSource": "local"
+        }
+      }
+    }
+  }
+  ```
+
+Usually, developers tend to provide CI jobs with secrets by using environment variables. One of the challenges with this apporach is that the `credentials.json` file contains a JSON object and it might be problematic to escape it properly so you could assign it to an environment variable. One of the solutions to this problem is to convert the file to a base64-encoded string, set an environment variable to that value, and later decode it and restore the file on the CI.
```

Comment:
```suggestion
Developers tend to provide CI jobs with secrets by using environment variables. One of the challenges with this approach is that the `credentials.json` file contains a JSON object and it might be difficult to escape it properly, so you could assign it to an environment variable. One of the solutions to this problem is to convert the file to a base64-encoded string, set an environment variable to that value, and later decode it and restore the file on the CI.
```